test/gtest/mha_find20.cpp

```diff
@@ -0,0 +1,395 @@
+
+            arguments[i].id         = it.first;
+            arguments[i].descriptor = &descVector[i];
+            arguments[i].buffer     = it.second->gpuBuffer.get();
+
+            ++i;
+        }
+
+        std::vector<miopenTensorArgumentId_t> output_ids = {miopenTensorMhaO,
+                                                            miopenTensorMhaAmaxO,
+                                                            miopenTensorMhaAmaxS,
+                                                            miopenTensorMhaM,
+                                                            miopenTensorMhaZInv};
+
+        std::size_t workspace_size = 0;
+        uint64_t solver_id;
+
+        Workspace workspace;
+
+        for(const auto& solution : solutions)
+        {
+            miopenGetSolutionWorkspaceSize(solution, &workspace_size);
+            miopenGetSolutionSolverId(solution, &solver_id);
+
+            workspace.resize(workspace_size);
+
+            std::cerr << "Run a solution." << std::endl;
+            EXPECT_EQUAL(miopenRunSolution(&handle,
+                                           solution,
+                                           numTensors,
+                                           arguments.get(),
+                                           workspace.ptr(),
+                                           workspace.size()),
+                         miopenStatusSuccess);
+
+            std::cerr << "Solution executed." << std::endl;
+
+            TensorStructMap outputTensorResults;
+
+            // reading results from find 2.0 call and preparing structures for non find 2.0 outputs
+            // to compare with
+            for(const auto& id : output_ids)
+            {
+                const TensorStructPtr& tensorStructPtr = tensors[id];
+                tensorStructPtr->tensorFloat.data      = handle.Read<float>(
+                    tensorStructPtr->gpuBuffer, tensorStructPtr->tensorFloat.data.size());
+
+                TensorStructPtr& ptr = outputTensorResults[id];
+
+                ptr              = TensorStructPtr(new TensorStruct());
+                ptr->tensorFloat = tensorStructPtr->tensorFloat;
+                ptr->gpuBuffer   = handle.Write(ptr->tensorFloat.data);
+            }
+
+            std::cerr << "Run via solver infrastructure directly." << std::endl;
+            GetResultsWithoutFind20(handle, outputTensorResults, workspace, solver_id);
+            std::cerr << "Run via solver infrastructure executed!" << std::endl;
+
+            for(const auto& id : output_ids)
+            {
+                double error = miopen::rms_range(outputTensorResults[id]->tensorFloat.data,
+                                                 tensors[id]->tensorFloat.data);
+                const double tolerance = 1e-3;
+
+                EXPECT_TRUE(std::isfinite(error) && error <= tolerance)
+                    << "Outputs do not match each other. Error:" << error;
```

Comment:
```suggestion
                EXPECT_TRUE(std::isfinite(error));
                EXPECT_LE(error, tolerance);
```
`EXPECT_LE` prints both values in case of condition violation.